File: gaia-sdk-javascript/src/graphql/request/type/Skill.ts

import {SkillVersion} from "./SkillVersion";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Represents Skill information
 */
export class Skill extends Array<(_:VariableRegistry) => string> {
public _typeName = "Skill";
    /**
     * Id of the tenant
     */
    public tenantId = () => { 
        this.push(_ => "tenantId")
    };

    /**
     * Skill reference
     */
    public reference = () => { 
        this.push(_ => "reference")
    };

    /**
     * The name of the skill
     */
    public qualifier = () => { 
        this.push(_ => "qualifier")
    };

    /**
     * Detailed description about the skill
     */
    public appendent = () => { 
        this.push(_ => "appendent")
    };

    /**
     * The list of labels of the skill
     */
    public labelList = () => { 
        this.push(_ => "labelList")
    };

    /**
     * The uri of the repository where the skill is
     */
    public repositoryUri = () => { 
        this.push(_ => "repositoryUri")
    };

    /**
     * The list of available and build skill versions
     */
    public versions = (config: (_:SkillVersion) => void) => this.push((registry) => {
        const entity = new SkillVersion();
        config(entity);
        return "versions { " + entity.render(registry) + " }";
    });

    /**
     * A list of all available version tags
     */
    public tags = () => { 
        this.push(_ => "tags")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/Skill.py

from gaia_sdk.graphql.request.type.SkillVersion import SkillVersion

from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class Skill(list):
    """
    Represents Skill information
    """

    """
    Id of the tenant
    """
    def tenant_id(self):
        self.append(lambda x: "tenantId")

    """
    Skill reference
    """
    def reference(self):
        self.append(lambda x: "reference")

    """
    The name of the skill
    """
    def qualifier(self):
        self.append(lambda x: "qualifier")

    """
    Detailed description about the skill
    """
    def appendent(self):
        self.append(lambda x: "appendent")

    """
    The list of labels of the skill
    """
    def label_list(self):
        self.append(lambda x: "labelList")

    """
    The uri of the repository where the skill is
    """
    def repository_uri(self):
        self.append(lambda x: "repositoryUri")

    """
    The list of available and build skill versions
    """
    def versions(self, config: Callable[['SkillVersion'], None]):
        def callback(registry: VariableRegistry):
            entity = SkillVersion()
            config(entity)
            return "versions {" + entity.render(registry) + "}"
        self.append(callback)

    """
    A list of all available version tags
    """
    def tags(self):
        self.append(lambda x: "tags")

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
